src/main/java/org/apache/commons/jxpath/ClassFunctions.java

```diff
@@ -92,6 +93,28 @@ public Function getFunction(
         String namespace,
         String name,
         Object[] parameters) {
+        return getFunction(namespace, name, parameters, null);
+    }
+
+    public Function getFunction(
+            String namespace,
+            String name,
+            Object[] parameters,
+            JXPathFilter jxPathFilter) {
+
+        // give chance to ClassFilter to filter out, if present
+        try {
+            if (jxPathFilter != null && !jxPathFilter.exposeToXPath(functionClass.getName())) {
+                throw new ClassNotFoundException(functionClass.getName());
```

Comment:
We shouldn't be using exceptions for control flow here, let's throw JXPathException directly.

Reply:
Good point! Changed to throwing a JXPathException directly.